airbyte-ci/connectors/pipelines/pipelines/models/singleton.py

```diff
@@ -0,0 +1,23 @@
+from typing import Any, Type
+
+
+class Singleton:
```

Comment:
Suggest adding test cases here like I added in this PR:

https://github.com/airbytehq/aircmd/pull/16

notably these test cases surface a small issue with my implementation that is almost certainly not applicable to your use case - if you load a module using Singleton dynamically (such as in the plugin use case) it is theoretically possible to instantiate multiple copies of Singleton. I would take a look at the failing test case and just ensure it will never apply to your use case or push a fix.